### PR TITLE
DAS-160 : Provide an option to remove data staging when restoring

### PR DESCRIPTION
**Pull request by swsachith**

### CODE REVIEW

File: components/analytics-tools/org.wso2.carbon.analytics.tools.backup/src/main/java/org/wso2/carbon/analytics/tools/backup/AnalyticsDataBackupTool.java [hunk 1]
```diff
@@ -180,7 +181,7 @@ public static void main(String[] args) throws Exception {
             } else if (line.hasOption(BACKUP_FILE_SYSTEM)) {
                 backupFileSystem(analyticsFileSystem, tenantId, baseDir);
             } else if (line.hasOption(RESTORE_RECORD_STORE)) {
-                restoreRecordStore(service, tenantId, baseDir, timeFrom, timeTo, specificTables);
+                restoreRecordStore(service, tenantId, baseDir, timeFrom, timeTo, specificTables,disableStaging);
```
Comment (chanikag) [inline components/analytics-tools/org.wso2.carbon.analytics.tools.backup/src/main/java/org/wso2/carbon/analytics/tools/backup/AnalyticsDataBackupTool.java]: Add a space between parameters

Comment (swsachith) [inline components/analytics-tools/org.wso2.carbon.analytics.tools.backup/src/main/java/org/wso2/carbon/analytics/tools/backup/AnalyticsDataBackupTool.java]: Fixed with the PR : https://github.com/wso2/carbon-analytics/pull/137

File: components/analytics-tools/org.wso2.carbon.analytics.tools.backup/src/main/java/org/wso2/carbon/analytics/tools/backup/AnalyticsDataBackupTool.java [hunk 2]
```diff
@@ -253,7 +254,15 @@ private static void restoreTable(AnalyticsDataService service, int tenantId, Str
                 System.out.println(myDir.getAbsolutePath() + " is not a directory to contain table data, skipping.");
                 return;
             }
-            AnalyticsSchema schema = readTableSchema(baseDir.getAbsolutePath() + File.separator + table);
+
+            // enabling/disabling the staging for the table
+            AnalyticsSchema currentSchema = readTableSchema(baseDir.getAbsolutePath() + File.separator + table);
+            AnalyticsSchema schema;
+            if (disableStaging)
```
Comment (sinthuja) [inline components/analytics-tools/org.wso2.carbon.analytics.tools.backup/src/main/java/org/wso2/carbon/analytics/tools/backup/AnalyticsDataBackupTool.java]: If condition needs to have {} for the better readability.

File: components/analytics-tools/org.wso2.carbon.analytics.tools.backup/src/main/java/org/wso2/carbon/analytics/tools/backup/AnalyticsDataBackupTool.java [hunk 3]
```diff
@@ -578,6 +587,29 @@ private static void createDirectoryInLocalSystem(String path) {
     }
 
     /**
+     * Returns an indexing disabled schema from the provided Analytics Schema.
```
Comment (sinthuja) [inline components/analytics-tools/org.wso2.carbon.analytics.tools.backup/src/main/java/org/wso2/carbon/analytics/tools/backup/AnalyticsDataBackupTool.java]: Explain the parameter in the java docs, have the space between the method comment to parameters' comments.

Comment (swsachith) [inline components/analytics-tools/org.wso2.carbon.analytics.tools.backup/src/main/java/org/wso2/carbon/analytics/tools/backup/AnalyticsDataBackupTool.java]: fixed with https://github.com/wso2/carbon-analytics/pull/137

File: components/analytics-tools/org.wso2.carbon.analytics.tools.backup/src/main/java/org/wso2/carbon/analytics/tools/backup/AnalyticsDataBackupTool.java (hunk 2 above)
Comment (swsachith) [inline components/analytics-tools/org.wso2.carbon.analytics.tools.backup/src/main/java/org/wso2/carbon/analytics/tools/backup/AnalyticsDataBackupTool.java]: fixed with https://github.com/wso2/carbon-analytics/pull/137

**Pull request merged**